.github/workflows/quick.yaml

```diff
@@ -16,6 +16,12 @@ on:
     # test PRs targeting this branch code
     branches: [ "master" ]
 
+concurrency:
+  # Cancel ongoing tests in case of push to the same PR or staging branch,
+  # but let previous master commit tests complete.
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
 env:
```

Comment:
Please add an equivalent optimization to slow.yaml (where it will only apply to staged commits).

Reply:
> > > > Please add an equivalent optimization to slow.yaml (where it will only apply to staged commits)

> > > Is it useful?

> > I do my best to avoid suggesting something I consider useless. Moreover, this PR itself says aborting staged commit tests is useful (AFAICT).

> All of us do; sometimes we all can be wrong (some more than others, but still)

While we all make mistakes, including a mistake of suggesting useless changes, I doubt asking an "is [your suggestion] useful?" question is a good way to make progress. Assuming that a collaborator believes they are suggesting something useful is often the best starting point, even if they are wrong! This is not a big deal, of course. I am only writing this in hope to avoid similar rhetorical questions/conversations in the future.